web/src/views/InvestigationDetail.vue

```diff
@@ -498,17 +505,84 @@ export default defineComponent({
       }));
     }
 
+    async function toggleNonTiledImageOverlay(pin: Pin, dataset: Dataset) {
+      if (!dataset.content) return;
+      if (!featureLayer) return; // sanity check. this should exist
+      if (!nonTiledOverlayFeature) {
+        nonTiledOverlayFeature = featureLayer.createFeature('quad');
+      }
+      const quadData = nonTiledOverlayFeature.data() || [];
+      // eslint-disable-next-line
+      const existingOverlay = quadData.find((overlay: any) => overlay.pinId === pin.id);
+      if (existingOverlay) {
+        nonTiledOverlayFeature.data(
+          // eslint-disable-next-line
+          nonTiledOverlayFeature.data().filter((overlay: any) => overlay.pinId !== pin.id),
+        ).draw();
+        nonTiledOverlayFeature.draw();
+      } else {
+        try {
+          const urlRoot = process.env.VUE_APP_API_ROOT;
+          const url = `${urlRoot}/datasets/tile_source/${dataset.id}/thumbnail.png`;
+          const image: HTMLImageElement = await getNonTiledImage(url);
+          image.crossOrigin = 'use-credentials';
+          if (store.state.axiosInstance) {
+            const imageMetadata = await (await store.state.axiosInstance.get(`${urlRoot}/datasets/tile_source/${dataset.id}/metadata`)).data;
+            const ul = postGisToPoint(pin.child_location) || { x: 0, y: 0 };
+            const lr = {
+              x: ul.x + (imageMetadata.sizeX || 0),
+              y: ul.y + (imageMetadata.sizeY || 0),
+            };
+            quadData.push({
+              ul,
+              lr,
+              image,
+              pinId: pin.id,
+            });
+            nonTiledOverlayFeature.data(quadData).draw();
+          }
+        } catch (error) {
+          // eslint-disable-next-line
+          console.error(`Failed to show overlay for dataset ${dataset.id}.`);
+        }
+      }
+    }
+
+    function toggleDatasetPin(pin: Pin) {
+      const childDataset: Dataset | undefined = store.state.currentDatasets.find(
+        (dataset: Dataset) => dataset.id === pin.child,
+      );
+      if (!childDataset) return;
+      switch (childDataset.dataset_type) {
+        case 'non_tiled_image':
+          toggleNonTiledImageOverlay(pin, childDataset);
+          break;
+        default:
+          break;
+      }
+    }
+
     watch(selectedPins, (newPins, oldPins) => {
```

Comment:
Why `watch` the selection state, rather than attach logic to the toggle state per pin?

(This is really an architecture question--I'm not necessarily suggesting any changes here.)

Reply:
Selecting/deselecting pins is done in the `PinList` component. Instead of bubbling up events to the `InvestigationDetail`, we're using the store to keep track of which pins the user has selected/deselected, so the `InvestigationDetail` can just monitor this state via a computed property and a watcher. Let me know if I'm answering your question or not.